editor/components/colors/with-colors.js

```diff
@@ -14,6 +14,40 @@ import { withSelect } from '@wordpress/data';
  */
 import { getColorValue, getColorClass, setColorValue } from './utils';
 
+const memoizedGetColor = memoize(
+	( colors ) =>
```

Comment:
There's a potential memory leak here when `getEditorSettings().colors` is not defined. In the `withSelect` below, you pass `[]` as the `defaultValue` in the `get` call. Since `[] !== []`, these will be stored as separate entries in the memoization cache.

```
$ n_
n_ > var fn = () => console.log( 'called' );
undefined
n_ > var memoized = _.memoize( fn );
undefined
n_ > memoized( [] );
called
undefined
n_ > memoized( [] );
called
undefined
n_ > memoized.cache.size
2
```

Reply:
Nice catch 👍